src/glang/lib/std/io.c3

```diff
@@ -5,46 +5,67 @@
 @require_once "iterators.c3"
 @require_once "math.c3"
 @require_once "string.c3"
+@require_once "sys/mman.c3"
+@require_once "sys/stat.c3"
 @require_once "sys/uio.c3"
 @require_once "syscalls.c3"
 @require_once "wrappers.c3"
 
-typedef UnbufferedFile : { fd : int }
+typedef File : { fd : int }
+typedef MappedFile : { underlying : File, buffer : View<u8> }
 
-function write : (file : UnbufferedFile&, buffer : View<u8>) -> void = {
+function write : ( file : File&, buffer : View<u8> ) -> void = {
     sys_write(file.fd, &buffer.data, buffer.length);
 }
 
-function read : (file : UnbufferedFile&, buffer: Span<u8>) -> View<u8> = {
+function read : ( file : File&, buffer: Span<u8> ) -> View<u8> = {
     let bytes_read: isize = sys_read(file.fd, &mut buffer.data, buffer.length);
     return {&buffer.data, bytes_read};
 }
 
-function close : (file : UnbufferedFile&) -> void = {
+function close : ( file : File& ) -> void = {
     if file.fd >= 3 {
         // Don't close std{in,out,err}
-        runtime_assert(sys_close(file.fd) > 0);
+        runtime_assert(sys_close(file.fd) >= 0);
```

Comment:
Actually, does `close()` ever return `> 0`?

Reply:
It doesn't... Guess we just weren't calling it before hmmm

Reply:
Oops!